lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
+
+		this.setState( { editorState }, () => {
+			if ( contentHasChanged ) {
+				this.props.onChangeContent( plainTextContent( editorState ) );
+			}
+		} );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			// if we've received this content before there's nothing to do now
+			nextProps.content !== this.props.content
+			// if the new content isn't different from what's rendered
+			// don't change it because we'd lose the SelectionState
+			&& plainTextContent( this.state.editorState ) !== nextProps.content
+		) {
+			const contentState = ContentState.createFromText( nextProps.content, '\n' );
+			this.setState( {
+				editorState: EditorState.createWithContent( contentState, urlDecorator )
+			} );
+		}
+	},
```

Comment:
This method is well-worth considering. What does it mean to have the props change and what does it mean to create a new `EditorState` object. I think that for one, we will lose our undo history. Do we want to be able to undo changes that pour in from outside of this particular client (for example, I make a change on my phone). I'm sure we could find a way to convey that information and use `EditorState.push()` instead with a specific action type, such as `'update-from-simperium'`.

Further, could we map the Simperium deltas into individual editor change sets such as found in the `Modifier` module. I'm sorry that I don't know these things off of the top of my head, such as if we even have access here to the Simperium diff.